File: minseo/week3/N개의 최소공배수/N개의 최소공배수.js
function solution(arr) {
    let result = 1;
    for(let i = 0; i<arr.length; i++){
        result = lcm(result, arr[i])
    }
    return result;
}

function gcd(a, b) {
  return b === 0 ? a : gcd(b, a % b);
}

function lcm(a, b) {
  return (a * b) / gcd(a, b);
}

/* 처음에 최대공약수와 최소공배수를 구하는 함수를 soluton 위에 썼더니, 시간초과가 걸렸지만, 밑에 쓰니 해결되었다. */

File: nahyun/week3/12953. N개의 최소공배수/N개의 최소공배수.java
import java.io.*;
import java.util.*;

class Solution {
    public static int solution(int[] arr) {
        int Length = arr.length;
        int answer = 0;
        for(int i = 0; i < Length-1; i++) {
            int gcd = GCD(arr[i], arr[i+1]);
            answer = arr[i] * arr[i+1] / gcd;
            arr[i+1] = answer;
        }
        return answer;
    }
    
    //최소 공약수
    public static int GCD(int A, int B) {
        if(B == 0) return A;
        else return GCD(B, A % B);
    }
    
}
